From 00e72ad6c139c0bb63e71d59d14f04cddde3a085 Mon Sep 17 00:00:00 2001
From: Paul Hunt <hunt@ti.com>
Date: Fri, 18 Mar 2011 14:59:03 -0600
Subject: syslink: ipu_pm: change target power state for IVAHD

Change the targeted next power state for IVAHD to Inactive
per the design. When the IVAHD state transition control mode
is set to HW_AUTO when the codec starts up, this must be
configured to On or Inactive or it will crash when the end
of the frame processing is done.

Change-Id: Ib61649f1e641142b786951bd8f80d5e0b162b30f
Signed-off-by: Paul Hunt <hunt@ti.com>
---
 drivers/dsp/syslink/ipu_pm/ipu_pm.c |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/drivers/dsp/syslink/ipu_pm/ipu_pm.c b/drivers/dsp/syslink/ipu_pm/ipu_pm.c
index b4a2a7e..1ff487d 100644
--- a/drivers/dsp/syslink/ipu_pm/ipu_pm.c
+++ b/drivers/dsp/syslink/ipu_pm/ipu_pm.c
@@ -66,6 +66,9 @@
 /* Module headers */
 #include <plat/ipu_dev.h>
 #include "../../../../arch/arm/mach-omap2/cm.h"
+#include "../../../../arch/arm/mach-omap2/prm.h"
+#include "../../../../arch/arm/mach-omap2/prcm-common.h"
+#include "../../../../arch/arm/mach-omap2/prm44xx.h"
 #include "ipu_pm.h"
 
 /** ============================================================================
@@ -1283,6 +1286,11 @@ static inline int ipu_pm_get_iva_hd(struct ipu_pm_object *handle,
 		pr_err("%s %d Error requesting IVA_HD\n", __func__, __LINE__);
 		return PM_UNSUPPORTED;
 	}
+	/* set Next State to INACTIVE for IVAHD */
+	prm_write_mod_reg(0xff0e02,
+			 OMAP4430_PRM_IVAHD_MOD,
+			 OMAP4_PM_IVAHD_PWRSTCTRL_OFFSET);
+
 	params->pm_iva_hd_counter++;
 	pr_debug("Request IVA_HD\n");
 
-- 
1.6.0.4

